### ISSUE TITLE
[ISSUE] Cannot create api key

### ISSUE THREAD

**Issue by ZIM555** [WillyXJ/facileManager#507]
Please prefix your issue title with one of the following: [BUG], [ISSUE], [FEATURE REQUEST], [MODULE REQUEST], [OTHER].

Replace everything between stars with current version of your facileManager and module installations:  
fM Version : **4.0.1**
{Module Name} Version : **4.0.1**

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [X] I have read and understood the [contributors guide](https://github.com/WillyXJ/facileManager/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [X] I have checked that the issue I'm posting isn't already solved and no duplicates exist in [closed issues](https://github.com/WillyXJ/facileManager/issues?q=is%3Aissue+is%3Aclosed) and [opened issues](https://github.com/WillyXJ/facileManager/issues)
- [X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

---

**(BUG | ISSUE) Expected Behavior:**

**(BUG | ISSUE) Actual Behavior:**

Could not create the keypair because a database error occurred.
Database error: [Field 'key_id' doesn't have a default value | Query: [INSERT INTO fm_keys (`account_id`, `user_id`, `key_token`, `key_secret`) VALUES (1, 1, 'RANDOMKEY', 'RANDOMKEY.')]]

**(BUG | ISSUE) Steps to reproduce:**

{replace this line with your issue content}

**Comment by ZIM555** [comment]
I do this request manual.
Where i can found how use API for exemple for adding records to domain or delete it?

**Comment by WillyXJ** [comment]
Thanks for the report. Somehow the AUTO_INCREMENT was missed in the install/upgrade code for the table schema. Until 4.0.2 is released, this can be fixed by running the following SQL statement against your database:

```
ALTER TABLE `fm_keys` MODIFY `key_id` int(11) NOT NULL AUTO_INCREMENT;
```

External documentation is still poor (#307), but the fmDNS client.php script's helpfile should provide what you need.

In short, add your API key and secret to config.inc.php:

```
define('APIKEY', 'MY_KEY');
define('APISECRET', 'MY_KEY_SECRET');
```

Then use the API via the client utility:

```
php /usr/local/facileManager/fmDNS/client.php setHost action=add domain=mydomain.com host=www value=1.2.3.4 type=A ttl=500 createPTR=yes comment="some comment" status=active
```

**Comment by ZIM555** [comment]
Thank you for answer. So for now i can use api only with 
fm client? 
Curl request to fmmanager not supported?
For now not possible use curl request for exemple to https://dnsmanagerdomain/api.php for update records on domain on all dns servers?

**Comment by WillyXJ** [comment]
Technically it can be done from any client that can POST data as that's what the fmDNS client does. However, the necessary variables to pass still needs to be documented in a nice format.

**Comment by ZIM555** [comment]
I have another question. Now i have this php test code
And they working. This code add A record to domain with id 21. But what i must do for automatically reload zone and apply changes? Because now i must go in panel and press reload manual. 

`<?php
function getPostData($url) {
	echo "getPostData<br>";
	$debug=true;
	$data = array(
					'module_type' => 'CLIENT',
					'AUTHKEY' => 'default', 
					'APIKEY' => 'mykey',
					'APISECRET' => 'mysecretkey',
					'module_name' => 'fmDNS',
					'domain_id' => '21',
					'api' => array(
							'action' => 'add',
							'record_type' => 'A',
							'record_value' => '1.2.3.4',
							'record_status' => 'disabled',
							'domain_id' => '21',
							'record_name' => 'test'
							)
						);

	$ch = curl_init();
	curl_setopt($ch, CURLOPT_URL, $url);
	curl_setopt($ch, CURLOPT_POST, true);
	curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($data));
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
	curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
	curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
	
	$result = curl_exec($ch);
	
	echo "Recive:<br>";
	var_dump($result);
	echo "<br>";
	
	if ($debug && $result === false) {
		echo "<br>" . curl_error($ch);
	}
	
	curl_close($ch);
	return $result;
}

echo "Begin<br>";
getPostData('http://domain/api.php');
?>
`

**Comment by WillyXJ** [comment]
Issuing a reload from the API is currently not supported.

**Comment by WillyXJ** [comment]
API key creation is now fixed in [fM 4.0.2](http://www.facilemanager.com/download/) and later.
